### PR TITLE
clang check Alignment

### PR DESCRIPTION
**Pull request by davidlange6**
Changes in Alignment subsystem due to clang-tidy if #21478 is merged

### CODE REVIEW

**Comment by davidlange6** [comment]
hum, those code checks seem unlikely to compile - the clang-changes must have caused problems elsewhere (that I didn't check for)

**Comment by davidlange6** [comment]
here is an example of the bug introduced

  if (newBareTkGeomPtr == bareTkGeomPtr_) { return; // already booked hists, nothing changed }       

end-of-line comments are not dealt with correctly.

File: Alignment/OfflineValidation/plugins/TrackerOfflineValidation.cc [hunk 1]
```diff
@@ -417,7 +417,7 @@ TrackerOfflineValidation::checkBookHists(const edm::EventSetup& es)
 {
   es.get<TrackerDigiGeometryRecord>().get( tkGeom_ );
   const TrackerGeometry *newBareTkGeomPtr = &(*tkGeom_);
-  if (newBareTkGeomPtr == bareTkGeomPtr_) return; // already booked hists, nothing changed
+  if (newBareTkGeomPtr == bareTkGeomPtr_) { return; // already booked hists, nothing changed }
```
Comment (fwyzard) [inline Alignment/OfflineValidation/plugins/TrackerOfflineValidation.cc]: I've tried to reproduce this wrong substitution with a minimal test case, and I haven't been able to.
What options did you use ?

**Comment by davidlange6** [comment]
sorry - I missed this mail - 

I effectively did

scram build code-checks USER_CODE_CHECKS_FILE=checks.txt

where checks.txt has ~all cc files in Alignment/* packages

and I added just readability-braces-around-statements to .clang-tidy.

<   ,readability-uniqueptr-delete-release,
<   ,readability-braces-around-statements,
---
>   ,readability-uniqueptr-delete-release


david

**Comment by fwyzard** [comment]
Mhm, strange.

I just ran
```bash
clang-tidy --fix Alignment/OfflineValidation/plugins/TrackerOfflineValidation.cc
```

and I got a valid file:
```diff
@@ -417,7 +424,8 @@ TrackerOfflineValidation::checkBookHists(const edm::EventSetup& es)
 {
   es.get<TrackerDigiGeometryRecord>().get( tkGeom_ );
   const TrackerGeometry *newBareTkGeomPtr = &(*tkGeom_);
-  if (newBareTkGeomPtr == bareTkGeomPtr_) return; // already booked hists, nothing changed
+  if (newBareTkGeomPtr == bareTkGeomPtr_) { return; // already booked hists, nothing changed
+}
 
   if (!bareTkGeomPtr_) { // pointer not yet set: called the first time => book hists
 
```

**Comment by davidlange6** [comment]
clang-tidy does behave differently when given multiple files, though I don't see a reason it would for this test (and this file, which is source file not a header file).. I'll try some other variations..

**Comment by davidlange6** [comment]
Your PR is unmergeable. Please have a look and possibly rebase it.

**Comment by arunhep** [comment]
@davidlange6 would you like to continue with these changes? Or if not, then may be you want to close it.
thanks.

**Comment by arunhep** [comment]
ping

**Comment by arunhep** [comment]
-1
to remove it from the list

File: Alignment/OfflineValidation/plugins/TrackerOfflineValidation.cc (hunk 1 above)
Comment (smuzaffar) [reply Alignment/OfflineValidation/plugins/TrackerOfflineValidation.cc]: @davidlange6 and @fwyzard this was due to cmssw build rules. We run clang-tidy with `-export-fixes` option and then post process the generated yaml files to 
- drop suggested change for files which are not part of PR change set
- remove duplicate changes for headers files

Problem is that when we write back the yaml file (using pyYAML) then it drops the extra new line e.g. if originally we have
```
Diagnostics:     
  - DiagnosticName:  readability-braces-around-statements
    Message:         statement should be inside braces
    FileOffset:      566
    FilePath:        /build/muz/cf/CMSSW_10_6_X_2019-04-11-2300/src/Alignment/CocoaModel/interface/MeasurementCOPS.h
    Replacements:    
      - FilePath:        /build/muz/cf/CMSSW_10_6_X_2019-04-11-2300/src/Alignment/CocoaModel/interface/MeasurementCOPS.h
        Offset:          566
        Length:          0
        ReplacementText: ' {'
      - FilePath:        /build/muz/cf/CMSSW_10_6_X_2019-04-11-2300/src/Alignment/CocoaModel/interface/MeasurementCOPS.h
        Offset:          591
        Length:          0
        ReplacementText: '
}'
```

then reading it and writing back generates
```
Diagnostics:     
  - DiagnosticName:  readability-braces-around-statements
    Message:         statement should be inside braces
    FileOffset:      566
    FilePath:        /build/muz/cf/CMSSW_10_6_X_2019-04-11-2300/src/Alignment/CocoaModel/interface/MeasurementCOPS.h
    Replacements:    
      - FilePath:        /build/muz/cf/CMSSW_10_6_X_2019-04-11-2300/src/Alignment/CocoaModel/interface/MeasurementCOPS.h
        Offset:          566
        Length:          0
        ReplacementText: ' {'
      - FilePath:        /build/muz/cf/CMSSW_10_6_X_2019-04-11-2300/src/Alignment/CocoaModel/interface/MeasurementCOPS.h
        Offset:          591
        Length:          0
        ReplacementText: ' }'
```

this causes the closing braces to be on the same line. This should be fixes once we have https://github.com/cms-sw/cmsdist/pull/4914 merged.

**Pull request closed without merging**